cpp/URLHostObjects.cpp

```diff
@@ -68,98 +82,129 @@ namespace fasturl {
                 return result;
             });
         }
-        if (key == "forEach") {
+        if (key == "forEach")
+        {
             return HOST_FN(rt, "forEach", 1, {
-                if (count != 1) {
+                if (count != 1)
+                {
                     return jsi::Value::undefined();
                 }
-                if (!args[0].isObject()) {
+                if (!args[0].isObject())
+                {
                     return jsi::Value::undefined();
                 }
                 auto callback = args[0].getObject(rt);
                 auto keys = _params.get_keys();
-                while (keys.has_next()) {
+                while (keys.has_next())
+                {
                     auto key = keys.next();
-                    if (key.has_value()) {
+                    if (key.has_value())
+                    {
                         auto key_str = key.value();
-                        auto value = _params.get(key_str).value();
+                        auto value_view = _params.get(key_str).value();
+
+                        // Create jsi::String directly from std::string_view
+                        jsi::String jsiValueStr = jsi::String::createFromUtf8(rt, std::string(value_view));
+                        jsi::String jsiKeyStr = jsi::String::createFromUtf8(rt, std::string(key_str));
+
                         callback.asFunction(rt).call(
```

Comment:
`forEach` should return `value` as the first parameter and `key` as the second, this code was calling the callback with an array containing both properties.